community/kernel/src/main/java/org/neo4j/kernel/api/exceptions/Status.java

```diff
@@ -400,7 +400,7 @@ public int hashCode()
     public enum Classification
     {
         /** The Client sent a bad request - changing the request might yield a successful outcome. */
-        ClientError( TransactionEffect.NONE,
+        ClientError( TransactionEffect.ROLLBACK,
```

Comment:
Current behaviour is different from what you describe. Whenever one statement fails with ClientError next statements will be accepted and seemingly successful but the whole transaction will never be able to commit because the underlying KTI was marked as failed.
This PR makes this behaviour just more explicit by rolling back transaction at once.

Do you think this is a correct thing to do or we should just not mark KTI as failed for ClientErrors?